contrib/system_containers/centos/config.json.template

```diff
@@ -423,5 +423,6 @@
             "source": "/proc",
             "type": "proc"
         }
+        $ADDTL_MOUNTS
```

Comment:
this looks awkward. One should start `$ADDTL_MOUNTS` with a comma to not break the json syntax :(

Reply:
what is the format of this `$ADDTL_MOUNTS`?

Reply:
I believe:
```
,
         {
             "destination": "/host",
             "options": [
                 "rbind",
                 "rw",
                 "mode=755"
             ],
             "source": "/",
         }
```
look at the first command and always remember to not put any command lastly (it's pretty bad IMHO)

Reply:
What happens if the `$ADDTL_MOUNTS` is empty?

Reply:
@kwoodson Correct, it's an empty line and doesn't impact the ``config.json``.

Reply:
> What happens if the `$ADDTL_MOUNTS` is empty?

What is rendering these templates?  POSIX provides [some options][1] for parameter expansion depending on whether the parameter is set nonempty, set empty, or unset.  But I'm not clear on whether those would be legal in the template.

And if it was me scripting some JSON munging, I'd prefer to use `jq` instead of shell templates.

[1]: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02

Reply:
> this looks awkward. One should start $ADDTL_MOUNTS with a comma to not break the json syntax :(

@runcom it is a bit awkward and requires the first item to have a starting ``,``, however I *think* having the ``,`` there by default with no ``$ADDTL_MOUNTS`` would cause an error. I have not verified this though.

Reply:
> @runcom it is a bit awkward and requires the first item to have a starting ,, however I think having the , there by default with no $ADDTL_MOUNTS would cause an error. I have not verified this though.

that will cause an error of course. I'm saying this is not a great approach :confused: but I don't have anything better actually :)

Reply:
@wking I believe it's ``python``'s ``string.Template`` in the ``atomic`` command.

Reply:
I agree it is ugly :-( but it is what we can do with what we have now.

In any case, I'd expect nobody to pass this configuration manually, the openshift-installer has already some nice way to define additional mounts using this pattern, e.g. for node:

```
openshift_node_syscon_auth_mounts_l:
- type: bind
  source: "{{ oreg_auth_credentials_path }}"
  destination: "/root/.docker"
  options:
  - ro
  - bind
```

Reply:
> I believe it's python's `string.Template` in the `atomic` command.

Hmm, maybe that's [here][1] and [here][2].  Docs for `string.Template` are [here][3], and POSIX's conditional expansion syntaxes are not supported.  So I agree with everyone else that this is ugly, but that I can't think of anything better ;).

[1]: https://github.com/projectatomic/atomic/blob/v1.19.1/Atomic/syscontainers.py#L289-L291
[2]: https://github.com/projectatomic/atomic/blob/v1.19.1/Atomic/util.py#L1037-L1039
[3]: https://docs.python.org/3/library/string.html#template-strings

Reply:
Agreed. There may be some cleaner ways we could re-architect system containers use of ``config.json`` for this kind of thing but that would be way beyond the scope of what's needed by the openshift folks atm/.